src/GafferImageUI/ImageView.cpp

```diff
@@ -316,34 +1191,130 @@ class ImageView::ColorInspector : public boost::signals::trackable
 			m_sampler->imagePlug()->setInput( m_deleteContextVariables->outPlug() );
 			m_sampler->pixelPlug()->setInput( m_pixel->outPlug() );
 
-			plug->getChild<Color4fPlug>( "color" )->setInput( m_sampler->colorPlug() );
+			evaluatorPlug->getChild<Color4fPlug>( "pixelColor" )->setInput( m_sampler->colorPlug() );
+
+
+			m_regionSampler->inPlug()->setInput( m_deleteContextVariables->outPlug() );
+			m_regionSampler->areaPlug()->setInput( m_region->outPlug() );
+			evaluatorPlug->getChild<Color4fPlug>( "regionColor" )->setInput( m_regionSampler->averagePlug() );
 
 			ImageGadget *imageGadget = static_cast<ImageGadget *>( m_view->viewportGadget()->getPrimaryChild() );
 			imageGadget->channelsChangedSignal().connect( boost::bind( &ColorInspector::channelsChanged, this ) );
+
+
+			// ---- Create a plug on ImageView for storing colorInspectors
+			plug->addChild( new ArrayPlug( "inspectors", Plug::In, new ColorInspectorPlug(), 1, 1024, Plug::Default & ~Plug::AcceptsInputs ) );
+			colorInspectorsPlug()->childAddedSignal().connect( boost::bind( &ColorInspector::colorInspectorAdded, this, ::_2 ) );
+			colorInspectorsPlug()->childRemovedSignal().connect( boost::bind( &ColorInspector::colorInspectorRemoved, this, ::_2 ) );
+
+			colorInspectorsPlug()->getChild<ColorInspectorPlug>( 0 )->modePlug()->setValue( 2 );
+
+			view->plugSetSignal().connect( boost::bind( &ColorInspector::plugSet, this, ::_1 ) );
 		}
 
 	private :
 
-		Plug *plug()
+		Gaffer::ArrayPlug *colorInspectorsPlug()
 		{
-			return m_view->getChild<Plug>( "colorInspector" );
+			return m_view->getChild<Plug>( "colorInspector" )->getChild<ArrayPlug>( "inspectors" );
+		}
+
+		const Gaffer::ArrayPlug *colorInspectorsPlug() const
+		{
+			return m_view->getChild<Plug>( "colorInspector" )->getChild<ArrayPlug>( "inspectors" );
+		}
+
+		void plugSet( Gaffer::Plug *plug )
+		{
+			if( plug->parent()->parent() == colorInspectorsPlug() )
+			{
+				ColorInspectorPlug *colorInspector = IECore::runTimeCast<ColorInspectorPlug>( plug->parent() );
```

Comment:
The point of `runTimeCast` is that it returns `nullptr` if the types don't match, but there are a _lot_ of calls in this file that don't check for the null case. Either we're confident it'll be what we want, in which case `static_cast` expresses that, or we're not, in which case we should deal with the failure case. Here, I think `static_cast` is probably fine? There's also `assertedRunTimeCast`, which I've never really got into the habit of using.